integration_tests/src/main/python/window_function_test.py

```diff
@@ -455,3 +456,55 @@ def test_window_aggs_for_rows_collect_list():
             (partition by a order by b,c_int rows between CURRENT ROW and UNBOUNDED FOLLOWING) as collect_struct
         from window_collect_table
         ''')
+
+
+# Generates some repeated values to test the deduplication of GpuCollectSet.
+# And GpuCollectSet does not yet support struct type.
+_gen_data_for_collect_set = [
+    ('a', RepeatSeqGen(LongGen(), length=20)),
+    ('b', IntegerGen()),
+    ('c_int', RepeatSeqGen(IntegerGen(), length=15)),
+    ('c_long', RepeatSeqGen(LongGen(), length=15)),
+    ('c_time', RepeatSeqGen(DateGen(), length=15)),
+    ('c_string', RepeatSeqGen(StringGen(), length=15)),
+    ('c_float', RepeatSeqGen(FloatGen(), length=15)),
```

Comment:
Have you verified that there are NaNs in this?  We are doing data gen and the default weight for NaNs is only going to insert them about 5% of the time.  We configure the operation as ` NaNEquality.UNEQUAL` I really would like to know for sure that this is being tested and 15 items makes it feel a bit less likely that they will show up.

Reply:
I added extra data column (`FloatGen().with_special_case(math.nan, 200.0)`) into test, which produces a lot of NaN values.